src/components/App.js

```diff
@@ -2,29 +2,117 @@ import React, { Component } from 'react';
       result: '0',
+      operandStack: [],
+      operatorStack: [],
+      arithmetic: ['÷', 'x', '-', '+'],
+      calculation: ['AC', '+/-', '%'],
+      finished: false,
+      finishedTotal: null,
+      statusText: '',
     };
   }
 
+  doArithmetic = (operandStack, operatorStack) => {
+    let { total: prevTotal, next, operation } = this.state;
+    prevTotal = operandStack.shift();
+    next = operandStack.shift();
+    operation = operatorStack.shift();
+    let objTotal = calculate({ prevTotal, next, operation }, operation);
+    let i = 0;
+    while (i < operatorStack.length) {
+      prevTotal = objTotal.total;
+      next = operandStack[i];
+      operation = operatorStack[i];
+      objTotal = calculate({ prevTotal, next, operation }, operation);
+      i += 1;
+    }
+
+    return objTotal;
+  };
+
+  setTotal = ({ total }) => {
+    this.setState({
+      finishedTotal: `${total}`,
+      finished: true,
+      operandStack: [],
+      operatorStack: [],
+    });
+
+    return { doneStatusText: '', doneDisplayText: '0' };
+  };
+
+  updateDisplay = (statusText, result) => this.setState({ statusText, result });
+
   handleClick = btnName => {
-    this.setState({ result: btnName });
+    const {
+      operandStack,
+      operatorStack,
+      arithmetic,
+      calculation,
+      result,
+    } = this.state;
+    let { statusText } = this.state;
+    this.setState({ finished: false, finishedTotal: null });
+
+    let display;
+    if (!arithmetic.includes(btnName)) {
+      display = result === '0' || arithmetic.includes(result)
+        ? `${btnName}`
+        : `${result}${btnName}`;
+      statusText = `${statusText}${btnName}`;
+    } else {
+      operandStack.push(result);
+      operatorStack.push(btnName);
+      display = btnName;
+      statusText = `${statusText}${btnName}`;
+    }
+
+    if (calculation.includes(btnName)) {
+      const next = btnName === 'AC' ? 0 : 1;
+      const output = calculate(
+        { prevTotal: result, next, operation: btnName },
+        btnName,
+      );
+      const { doneStatusText, doneDisplayText } = this.setTotal(output);
+
+      statusText = doneStatusText;
+      display = doneDisplayText;
+    }
+
+    if (btnName === '=') {
+      operandStack.push(result);
+      const output = this.doArithmetic(operandStack, operatorStack);
+      const { doneStatusText, doneDisplayText } = this.setTotal(output);
+
+      statusText = doneStatusText;
+      display = doneDisplayText;
+    }
+
+    this.updateDisplay(statusText, display);
   };
 
   render() {
-    const { result } = this.state;
+    let { result, statusText } = this.state;
+    const { finished, finishedTotal } = this.state;
+    result = finished ? finishedTotal : result;
+    statusText = finished ? 'Ready' : statusText;
     return (
       <div className="app">
-        <Display result={result} />
+        <Display result={result} statusText={statusText} />
         <ButtonPanel clickHandler={this.handleClick} />
```

Comment:
it should pass TOTAL or NEXT to DIsplay component
https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png

Reply:
Yes. It passes total or next accordingly to the ```Display``` but with a variable name called ``result``
I am thiniking that requirement is talking about the value of _total_ or _next_